codegen/src/main/java/software/amazon/awssdk/codegen/model/config/customization/CustomizationConfig.java

```diff
@@ -123,6 +123,8 @@
 
     private Map<String, String> modelMarshallerDefaultValueSupplier;
 
+    private boolean useAutoConstructList = false;
```

Comment:
I've gotten into an existential crisis now. What do auto-construct lists mean for the customer and the service in this context?

Reply:
Customers shouldn't have to worry about auto constructed lists, which is why I kept the return types as `List<T>`. 

For services, the only one I'm aware of currently is CloudFormation where we explicitly distinguish between a normal empty list and an auto constructed one and only send the normal list variety.

```java
cfnRequestBuilder.withListMember(new ArrayList<>())
```
vs

```java
cfnRequestBuilder.withListMember(null)
```

and internally we instantiate an auto constructed list.


@shorea probably has more insight on this.

Reply:
ELB definitely has some weirdness too. We should test that. I want to say KMS had something weird with Maps but I can't exactly remember.

An auto constructed list is essentially null for marshalling purposes. So for a JSON service we would omit the field entirely from the document if it's an auto constructed list.

Reply:
Do you think we could get away without having CloudFormation being a special case and just default to empty lists for responses? It's annoying for a customer to distinguish between empty list and null for responses, but having so much code for just CloudFormation feels like a lot of work.

Reply:
Do we know of any services where there is a semantic difference between null and empty lists in responses? 

If the customer doesn't set anything on the request then we treat that as auto constructed list and send null on the wire (whatever that means for the protocol)
If the customer explicitly sets an empty list then we treat that as non-null and will send an empty list on the wire.

Inbound both null and empty lists will appear as empty lists to the customer with no real way to distinguish between them. If a service has a semantic difference between this (which they probably shouldn't) then we'd have to not auto construct for that service or perhaps just operation.